lib/jsdom/level1/core.js

```diff
@@ -245,7 +245,7 @@ core.DOMImplementation.prototype = {
 };
 
 core.Node = function Node(ownerDocument) {
-  EventTarget.call(this);
+  EventTarget.init.call(this);
```

Comment:
So, this is mostly just about avoiding having to copy args and also allow a prototype-based declaration approach to allow a bit of elegance in the implementation: Let's imagine having a generated `Event` and we do our own extensions from there (like we currently have with EventTarget).

```js
function EventImpl(type, initArgs) {
  // my init code
}

module.exports = {
  init: EventImpl, // nice trick, impossible with class-based declaration syntax though
  interface: EventImpl
};
```

Event has a constructor, so we need to provide it type and initArgs. So our MouseEvent class would look (simplified) like this:
```js
function MouseEvent(type, initArgs) {
  Event.init.call(this, type, initArgs);
}
```
init then just does:
```js
function init() {
  Impl.init.apply(this, arguments);
}
```


If we do function instead of method:
```js
function MouseEvent(type, initArgs) {
    Event.init(this, type, initArgs);
}
```
`Event.init`
```js
function init(obj) {
  const args = []; // copying not required in first approach
  for (let i = 1; i < arguments.length; ++i) {
     args[i - 1] = arguments[i];
  }
  Impl.init.call(this, type, initArgs);
}
```

Third choice would be to forward the additional arg all the way through to the `Impl.init` method:
```js
function MouseEvent(type, initArgs) {
    Event.init(this, type, initArgs);
}
```
`Event.init`
```js
function init(obj) {
  Impl.init.apply(null, arguments);
}
```
`Impl.init`
```js
function init(obj) { // explicit declaration not required in previous two approaches
  const args = []; // copying not required in first approach
  for (let i = 1; i < arguments.length; ++i) {
     args[i - 1] = arguments[i];
  }
  EventImpl.apply(obj, args);
}
```